### ISSUE TITLE
[Type] Feedback on typography and type hierarchy

### ISSUE THREAD

**Issue by mollieru** [18F/usfwds#262]
![type](https://cloud.githubusercontent.com/assets/3680494/9070911/3a99638c-3ac0-11e5-8ceb-4c3dcf321ffb.png)

**Comment by mollieru** [comment]
Feedback from @mebates:

-          Is there a consistent pattern to your leading? It looks tight on Display (what would happen if your second line has a capitalized letter?) We use a consistent 1.375 em calculation for our leading.
-          I think the H2 is reading as more heavy/important that H1. Noticed this specifically on the dhs_sample file
-          Are you specifying gray for H5? You’ll want to test that against all of your Background Colors for accessibility too. I would consider leaving color out of your type hierarchy (what color is my H5 if all of my type is navy?).
-          I would ask if the odd numbers are necessary (ex: 63 px vs 62 on Display). The more consistent and logical the hierarchy is, the easier it is to use
-          Body copy is pretty big! I know that’s the trend, but we might have a hard time with that on a copy dense pages (think less services, more explaining concepts) or blogs or press releases.

**Comment by Scotchester** [comment]
I'm a little concerned with the potential number of fonts that will be required and the performance implications of that.

Is the intention to use both the sans-serif and serif stacks on the same page/site? If so, we're looking at a webfont load of at least 10, by my count. Or at least 5 for sans and 5 for serif, if you don't combine them.

Sans: 300, 400, 500, 700, 400 italic
Serif: 300, 400, 700, 800, 800 italic

Plus, in both cases, any other italics in different weights that you might find necessary in the non-body styles.

Aside from pure performance implications, [this article that's been my webfont implementation gospel for a couple years](http://www.smashingmagazine.com/2013/02/setting-weights-and-styles-at-font-face-declaration/) indicates that using more than four in the style-linked technique will break IE7 and IE8, if we still care about those browsers.

(Separately, the style-linking technique breaks on iPad 1 and BlackBerry devices, so it may not be advisable anyway, in which case it's purely a performance concern.)

---

Agree with @mebates on wanting to see an example of a wrapping Display heading with a capital on the second line. Disagree that they should all be the same. Line height for body copy is far too large for display copy, and we don't actually keep it consistent, as she states :)

**Comment by maya** [comment]
I thought I remember on a previous thread a discussion about wanting to keep the number of fonts low?

**Comment by msecret** [comment]
I agree with @Scotchester, having to include two separate with various weights and sizes is a dangerous proposition.

The nature of the pattern library makes fonts more difficult to account for. On a standard website, it's easier to limit the amount of styles and weights of a font family that are needed based on the design, which allows the team to reduce the total amount of fonts downloaded. For instance, a designer might decide to ditch the Italic heading to save on performance after being consulted by his front end developer. For a pattern library, the design is more general, and is meant to be very flexible so developers can extend from the library for their needs. This would mean, from a developer user experience, we'd want to give them many font options to work with, which in turn means including more fonts to the point of it being a performance problem.

A way to get around this would be to design UI or developer interfaces to inform developers about the impact of fonts and allow them to limit the fonts based on their needs. I could imagine this being many different things, but one idea would be a UI where developers can select which fonts are required for their design when they download our bundle of files.

Building this pattern library puts us in a position of offering best practices on creating well-designed interfaces based on our user research. I think we should also push for best practices on performance, which would mean limiting the number of font choices.

**Comment by carodew** [comment]
@msecret – I think these are good points. I like how you're thinking about the scenarios in which the library could be used.

Another thing to consider is that one of the use cases for the library is the ability to use it as a framework, which seems like it would also benefit from looking at it holistically from a performance perspective.

**Comment by Scotchester** [comment]
Yes to all those thoughts! :+1:

**Comment by maya** [comment]
+1 to @msecret's thoughtful response!

**Comment by mebates** [comment]
A few other notes from my colleague @caheberer, looking specifically at the Data.gov example:

- What are the rules around using all-caps in the typography? In ours, the all caps options are defined as part of the hierarchy. On Data.gov, the header/nav hierarchy is hard to follow. A suggestion would be to limit the use of all caps to one type style.
- Lead Paragraph doesn’t feel different enough from h2. Maybe you could try a few more examples to make sure these two are different enough to lend them as separate styles or explore the relationship between Lead Paragraph and it’s proximity to other type styles.

**Comment by maya** [comment]
It sounds like if we could limit the fonts to 4 we'll be in better shape for performance. Is it possible to just use one font for Merriweather, like just for headlines? Then we can keep regular, bold, italics on Source Sans Pro. @mollieru @Scotchester @msecret @angel

**Comment by juliaelman** [comment]
Closing issue, as feedback has been completed and the work is in progress.
